### PR TITLE
Class methods

### LINKED ISSUE #20: implement Model#fill like Lucid

### PR DESCRIPTION
**Pull request by Xstoudi**
Should fix #20 and #21

### CODE REVIEW

File: src/Model/Model.ts [hunk 1]
```diff
@@ -313,6 +313,19 @@ export class Model {
     this.$isDeleted = true;
     return result.deletedCount === 1;
   }
+
+  public merge<T>(values: Omit<T, 'id' | ModelReadonlyFields>) {
+    Object.keys(values).forEach((key) => {
```
Comment (targos) [inline src/Model/Model.ts:318]: You can use `Object.entries` to have access to the value as well

File: src/Model/__tests__/Model.test.ts [hunk 2]
```diff
@@ -249,3 +249,34 @@ test('created user should not be dirty', async () => {
...
+
+  expect(user).toHaveProperty(['hello']);
+  expect((user as any).hello).toBe(true);
+});
+
+test('fill method', async () => {
+  const user = new User();
+  user.password = 'rootroot';
+
+  await user.fill({ username: nextUsername() }).save();
```
Comment (targos) [inline src/Model/__tests__/Model.test.ts:278]: What happens to the `createdAt` and `updatedAt` fields?

Comment (Xstoudi) [reply src/Model/__tests__/Model.test.ts:278]: Yeah I was working on it ^^

File: src/Model/__tests__/Model.test.ts [hunk 3]
```diff
@@ -249,3 +249,46 @@ test('created user should not be dirty', async () => {
...
+
+  expect(user.password).toBeUndefined();
+  expect(user.username).toBeDefined();
+});
+
+test('fill method after save', async () => {
+  const user = new User();
+  user.password = 'rootroot';
+  await user.save();
+  const createdAt = (user as any).createdAt;
```
Comment (targos) [inline src/Model/__tests__/Model.test.ts:288]: Why do you need to cast here?

Comment (Xstoudi) [reply src/Model/__tests__/Model.test.ts]: I dont, fixed.

**Review by targos — APPROVED** [review]

**Pull request merged**